buildSrc/subprojects/buildquality/src/main/kotlin/org/gradle/gradlebuild/buildquality/CiReportingPlugin.kt

```diff
@@ -25,38 +31,103 @@ import java.io.File
  * Team City.
  */
 open class CiReportingPlugin : Plugin<Project> {
-
     override fun apply(project: Project): Unit = project.run {
+        // Configure the testFilesCleanup policy in each subproject's build script
+        subprojects.forEach { it.extensions.create<TestFileCleanUpExtension>("testFilesCleanup", objects) }
+
         if (BuildEnvironment.isCiServer) {
             gradle.buildFinished {
-                prepareReportsForCiPublishing()
+                val failedTasks = project.failedTasks()
+                val executedTasks = project.executedTasks()
+                val tmpTestFiles = project.subprojects.flatMap { it.tmpTestFiles() }
+                prepareReportsForCiPublishing(failedTasks, executedTasks, tmpTestFiles)
+                project.cleanUp(tmpTestFiles.map { it.first })
```

Comment:
Should we cleanup only the `tmpTestFiles` for projects where we have failing tests? If not, verifyTestFilesCleanup will always succeed, right?

Reply:
Note that this `cleanUp()` is just to get rid of TeamCity warnings. This cleanup action doesn't affect the following `verifyTestFilesCleanup` because `tmpTestFiles` has already stored the leaked files.

Reply:
Ah, now I got. Thank you for the explanation.